### DISCUSSION TITLE
usemodel always return undefined

### DISCUSSION THREAD

**Discussion by hasakilol** [umijs/umi#12004] [General]
After upgrading umi 3 to umi 4, I begin to use `useModel`. My previous umi 3 model (many models in `src/models` directory) looks like: 
```
export default {
  namespace: 'user',
  state: {},
  effects: {},
  reducers: {}
}
```
I didn't change anything but only add `model: {}` to `config.ts`. 

Now when I call `useModel('user')`, it always return undefined. Any clue? Do I need to add some Provider anywhere?

**Comment by hasakilol** [comment]
It turns out `useModel` and `src/models` are different things 😓

**Comment by fz6m** [comment]
如有问题，需要给一个最小复现仓库。

你的这种 model 是 dva 的 model 吧：

```ts
export default {
  namespace: 'user',
  state: {},
  effects: {},
  reducers: {}
}
```

他需要 `dva: {}` ，不是 `useModel` ，`useModel` 是数据流的 model ，需要 `model: {}`

**Reply by hasakilol** [reply]
是的。我的是dva的model。 和 useModel 是两回事。

**Reply by hasakilol** [reply]
可以close了。
